frontend/src/views/TE/Upload.vue

```diff
@@ -0,0 +1,288 @@
+        :useCustomSlot="true"
+      >
+        <div>
+          <p>Drop files here to upload or click here to pick the files from your computer.</p>
+        </div>
+      </vue-dropzone>
+    </div>
+    <div>
+      <div :style="{ alignItems: 'center' }" class="d-flex pa-1">
+        <span v-if="totalUploadCount" class="pl-2">
+          Uploading {{ `${totalUploadCount} item${totalUploadCount > 1 ? 's' : ''}` }}
+          | {{ getTotalUploadTime()}}
+        </span>
+        <div :style="{ justifyContent: 'flex-end', display: 'flex' }">
+          <v-btn
+            v-if="completedFileUploads"
+            @click="clearCompletedFiles"
+            color="green"
+          >Clear completed</v-btn>
+          <v-btn v-if="totalUploadCount" @click="cancelAllFiles" color="red">Cancel all</v-btn>
+        </div>
+      </div>
+      <v-divider v-if="getFiles().length"></v-divider>
+      <v-list two-line>
+        <template v-for="[file, status] in getFiles()">
+          <div :key="file.upload.uuid">
+            <v-list-tile>
+              <v-list-tile-content>
+                <v-list-tile-sub-title
+                  :style="{ color: 'red' }"
+                  v-if="status.error"
+                >{{ status.error }}</v-list-tile-sub-title>
+                <v-list-tile-title>{{ file.name }}</v-list-tile-title>
+              </v-list-tile-content>
+              <v-list-tile-action :style="{ flexDirection: 'row' }">
+                <v-btn v-if="status.error" color="red" @click="deleteFile(file)">remove</v-btn>
+                <v-icon color="green" v-else-if="status.complete">fa-check-circle</v-icon>
+                <div v-else>
+                  <v-progress-circular
+                    :value="status.progress"
+                    color="green"
+                    :style="{ marginRight: '16px'}"
+                  ></v-progress-circular>
+                  <v-btn color="red" @click="cancelFile(status)" v-if="status.uploading">Cancel</v-btn>
+                </div>
+              </v-list-tile-action>
+            </v-list-tile>
+            <v-divider></v-divider>
+          </div>
+        </template>
+      </v-list>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+import { mapState } from "vuex";
+import VueDropzone from "vue2-dropzone";
+import "vue2-dropzone/dist/vue2Dropzone.min.css";
+import moment from "moment";
+import firebase from "firebase/app";
+import "firebase/storage";
+
+import { getTaskId, validateFlacFile } from "@/utility";
+
+interface IFileStatus {
+  progress?: number;
+  uploadTask?: any; // upload ref from firebase
+  error?: string;
+  complete?: boolean;
+  uploadStartedAt?: Date;
+  uploadRemainingTime?: number;
+  uploading?: boolean;
+  downloadUrl?: string;
+}
+
+const teUpload = firebase
+  .app()
+  .storage(
+    `${process.env.VUE_APP_TE_UPLOADS_SUB_DOMAIN}.${
+      process.env.VUE_APP_STORAGE_ROOT_DOMAIN
```

Comment:
I've changed this to `VUE_APP_PROJECT_DOMAIN` in https://trello.com/c/59mJXDBs/73-storage-structure, please adjust accordingly.